javascript/modules/client/src/internal/client-core.ts

```diff
@@ -166,4 +175,37 @@ export abstract class ClientCore implements IClientCore {
       }
     );
   }
+  public async pin(input: string | Uint8Array): Promise<string> {
+    if (!this.ipfs)
+      return Promise.reject(new Error("IPFS client is not initialized"));
+    return this.ipfs
+      .add(input)
+      .then(res => {
+        return Promise.resolve(res.path);
+      })
+      .catch(e => {
+        return Promise.reject(e);
+      });
+  }
+  public async fetchBytes(cid: string) {
+    if (!this.ipfs)
+      return Promise.reject(new Error("IPFS client is not initialized"));
+    try {
+      for await (const chunk of this.ipfs.cat(cid)) {
+        return Promise.resolve(chunk);
+      }
+    } catch (e) {
+      Promise.reject(e);
+    }
+    return;
+  }
+  public fetchString(cid: string): Promise<string> {
+    return this.fetchBytes(cid)
+      .then(bytes => {
+        return new TextDecoder().decode(bytes);
+      })
+      .catch(e => {
+        return Promise.reject(e);
+      });
```

Comment:
This code has no effect, I would either remove the catch block or throw a specific error.... but since nothing specific can fail here, I'd simply transparently relay the eventual error error